app/commands/v2/publish.rb

```diff
@@ -107,6 +107,11 @@ def set_first_published_at(content_item)
         content_item.update_attributes!(first_published_at: Time.zone.now)
       end
 
+      def pathless?(content_item)
+        !Location.find_by(content_item: content_item) &&
+          ContentItem::EMPTY_BASE_PATH_FORMATS.include?(content_item.schema_name || content_item.format)
```

Comment:
Could be less verbose to have a `requires_path?` method on content_item or in this class?